theatre/studio/src/utils/invariant.ts

```diff
@@ -0,0 +1,72 @@
+import {stringify} from './stringify'
+
+type accepts = string | number | object
+export function invariant(
```

Comment:
Needs docs. What is an invariant? It seems like this is some sort of an assertion? Is this for adding tests inline in our code? How is it used? I cannot exactly tell what sort of output this would produce in what cricumstances since the body of the function is quite general.